src/sign.c

```diff
@@ -1787,16 +1815,21 @@ sign_getlist(char_u *name, list_T *retlist)
             return;
     }
 
-    for (; sp != NULL && !got_int; sp = sp->sn_next)
+    while (sp != NULL && !got_int)
```

Comment:
I think that using `for` is more concise in this case.

Reply:
The way I usually see it is that you use a for loop only when you use all the arguments (a initialization, a guard and an increment). At least in my mind this helps distinguishing when I'm iterating a external variable because I will iterate with for loops only stuff I put in the arguments. Having said that I wont die on a hill for this so I can reverse it.

Reply:
Moving the increment to the end of the block makes it difficult to understand the loop, I think.

Additionally, if someone wants to use a continue inside the loop, it works differently.

Reply:
Done